### ISSUE TITLE
`--success last` does not work while `--success first` does

### ISSUE THREAD

**Issue by sergeyevstifeev** [open-cli-tools/concurrently#185]
**Expectation**
Switching the order of two commands while simultaneously switching from `--success last` to `--success first` should not change the exit status of `concurrently`

**Observed behaviour**
`--success last` does not work as expected when used together with `--kill-others`

**Version**
4.1.0

**Steps to reproduce**
Take two commands: `sleep 5; exit 0;` and `sleep 100`, and switch their order. With `--success first` works correctly:
```
$> concurrently --kill-others --success first 'sleep 5; exit 0;' 'sleep 100'
[0] sleep 5; exit 0; exited with code 0
--> Sending SIGTERM to other processes..
[1] sleep 100 exited with code SIGTERM
$> echo $?
0
```

With `--success last` works incorrectly (expected to get exit code `0` same as above):
```
$> concurrently --kill-others --success last 'sleep 100' 'sleep 5; exit 0;'
[1] sleep 5; exit 0; exited with code 0
--> Sending SIGTERM to other processes..
[0] sleep 100 exited with code SIGTERM
$> echo $?
1
```

**Comment by jcollum-nike** [comment]
Maybe this is what I'm seeing right now...

**Comment by jcollum-nike** [comment]
YEP! Changing the order of commands fixed the issue I was seeing. 

`4.1.0`

**Comment by wilgert** [comment]
We are running into the same problem, unfortunately we cannot switch the order of the commands.

**Comment by jcollum-nike** [comment]
maybe you need a pre script for the thing that definitely needs to happen first?

**Comment by RynatSibahatau** [comment]
I have the same issue for both first and last on 4.1.0. Seems there is no issue in 4.0.1.

**Comment by jcollum-nike** [comment]
I think this might be related https://github.com/kimmobrunfeldt/concurrently/issues/187

**Comment by gustavohenke** [comment]
Hello all! I'm sorry that I took this long to reply.

I think there's a misconception with how `--success` works.
From its description:

```
-s, --success     Return exit code of zero or one based on the success or
                  failure of the "first" child to terminate, the "last child",
                  or succeed only if "all" child processes succeed.
                            [choices: "first", "last", "all"] [default: "all"]
```

Please pay **special attention** to the `first child to terminate` part.
Although @sergeyevstifeev showed that he inverted the order of his commands, they are still exiting in the same order.

**Comment by gustavohenke** [comment]
Does the above help anyone? 🙂

**Comment by sergeyevstifeev** [comment]
@gustavohenke Ah, I see, this makes sense now. A bit hard to understand from the description, since "first" can be both time-wise and order-wise 🙂

**Comment by gustavohenke** [comment]
Maybe `first-specified`/`last-specified` (or better names!) could be made options for `--success`?

**Comment by jcollum-nike** [comment]
I'd like it if I could specify which process is the exit code I care about, `--exit=2` or something

**Comment by gustavohenke** [comment]
That SGTM too -- make `--success` accept a number, which refers to the process index.

**Comment by jcollum-nike** [comment]
I'd be happy with 

- `--success=2` means the 3rd item in the array is the one that sets the exit code 
- `--success=all`: any non-zero exit code will result in a non-zero exit code for the group of processes
- (not set): exit codes are irrelevant / ignored

**Comment by gustavohenke** [comment]
> * (not set): exit codes are irrelevant / ignored

Hmm that's a breaking change. The default is `all`.

Anyway, I'm happy to review PRs for this 🙂

**Comment by paescuj** [comment]
Implemented by #318 which has been released in [v7.2.0](https://github.com/open-cli-tools/concurrently/releases/tag/v7.2.0).
